src/storage/mutable_storage.hpp

```diff
@@ -51,26 +61,20 @@ struct mutable_storage
         return this + 1;
     }
 
-    /// Offset the storage
-    /// @param offset The number of bytes to offset the storage with
-    mutable_storage& operator+=(uint32_t offset)
+    /// @return data pointer
+    uint8_t* data() const
```

Comment:
should this be `const uint8_t* data() const` and then also have a `uint8_t* data()` member?